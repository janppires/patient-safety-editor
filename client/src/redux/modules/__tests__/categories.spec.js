import categoriesReducer, {
  addCategory,
  categoriesSelector,
  categorySelector
} from "redux/modules/categories";

describe("reducer: categories.byId ", () => {
  it("adds new category by id to empty state", () => {
    const category = { id: 2, name: "random name" };
    const action = addCategory(category);
    expect(categoriesReducer({}, action)).toMatchSnapshot();
  });

  it("adds new category by id to existing state", () => {
    const state = prepareState();
    const category = { id: 2, name: "random name" };
    const action = addCategory(category);
    expect(categoriesReducer(state, action)).toMatchSnapshot();
  });

  const prepareState = () => ({
    byId: {
      33: {
        id: 33,
        name: "other 33 category"
      }
    },
    list: [33]
  });
});

describe("categories selectors", () => {
  test("get categories array from state", () => {
    const state = prepareStateWithCategories();
    const categories = categoriesSelector(state);
    expect(categories).toMatchSnapshot();
  });

  test("get undefined category when categoryId is unknown", () => {
    const state = prepareStateWithCategories();
    const category = categorySelector(state);
    expect(category).toMatchSnapshot();
  });

  test("get category from id", () => {
    const state = prepareStateWithCategories();
    const categoryId = 2;
    const category = categorySelector(state, categoryId);
    expect(category).toMatchSnapshot();
  });

  const prepareStateWithCategories = () => ({
    categories: {
      list: [1, 2],
      byId: {
        1: { name: "random" },
        2: { name: "random 2" }
      }
    }
  });
});
