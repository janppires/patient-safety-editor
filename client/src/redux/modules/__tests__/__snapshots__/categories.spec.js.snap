// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories selectors get categories array from state 1`] = `
Array [
  Object {
    "name": "random",
  },
  Object {
    "name": "random 2",
  },
]
`;

exports[`categories selectors get category from id 1`] = `
Object {
  "name": "random 2",
}
`;

exports[`categories selectors get undefined category when categoryId is unknown 1`] = `undefined`;

exports[`reducer: categories.byId  adds new category by id to empty state 1`] = `
Object {
  "byId": Object {},
  "fetchingStatus": "CATEGORIES/FETCHED",
  "list": Array [],
}
`;

exports[`reducer: categories.byId  adds new category by id to existing state 1`] = `
Object {
  "byId": Object {
    "33": Object {
      "id": 33,
      "name": "other 33 category",
    },
  },
  "fetchingStatus": "CATEGORIES/FETCHED",
  "list": Array [
    33,
  ],
}
`;
